src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -43,23 +43,38 @@ public LogicManager(Model model, Storage storage) {
         addressBookParser = new AddressBookParser();
     }
 
+    /**
+     * Executes the command and returns the result.
+     *
+     * @param commandText The command as entered by the user.
+     * @return the result of the command execution.
+     * @throws CommandException If an error occurs during command execution.
+     * @throws ParseException   If an error occurs during parsing.
+     */
     @Override
     public CommandResult execute(String commandText) throws CommandException, ParseException {
-        logger.info("----------------[USER COMMAND][" + commandText + "]");
+        try {
+            logger.info("----------------[USER COMMAND][" + commandText + "]");
 
-        CommandResult commandResult;
-        Command command = addressBookParser.parseCommand(commandText);
-        commandResult = command.execute(model);
+            CommandResult commandResult;
+            Command command = addressBookParser.parseCommand(commandText);
+            commandResult = command.execute(model);
+
+            if (!commandResult.isPrompt()) {
+                model.clearSavedCommand();
+            }
 
-        try {
             storage.saveAddressBook(model.getAddressBook());
+
+            return commandResult;
         } catch (AccessDeniedException e) {
             throw new CommandException(String.format(FILE_OPS_PERMISSION_ERROR_FORMAT, e.getMessage()), e);
         } catch (IOException ioe) {
             throw new CommandException(String.format(FILE_OPS_ERROR_FORMAT, ioe.getMessage()), ioe);
+        } catch (Exception e) {
+            model.clearSavedCommand();
+            throw e;
```

Comment:
I feel like there is no need for this, if not perhaps you can throw a more specific exception.

Reply:
The reason that this is here is because I want savedCommands to be cleared regardless whether the next command is valid or invalid.
eg. 
1. delete i/123
2. invalidinput -> unknown command
3. confirm -> No command to confirm

since the execution of Commands without the flag hasPrompt, will result in clearing of the savedCommand, I also needed a way to clear the commands, if the input was invalid.

I agree that this approach is not good since im catching all exceptions and throwing it again. But I will think of better ways to handle this in the future.